File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < answer.length; i++) {
            if (!map.containsKey(toStr(s.charAt(i)))) {
                map.put(toStr(s.charAt(i)), i);
                answer[i] = -1;
            } else {
                answer[i] = i - map.get(toStr(s.charAt(i)));
                map.replace(toStr(s.charAt(i)), i);
            }
        }
        return answer;
    }
    
    private String toStr(char ch) {
        return String.valueOf(ch);
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let answer = [];
    let map = new Map();
    for (let i = 0; i < s.length; i++) {
        if (map.has(s.charAt(i))) {
            answer[i] = i - map.get(s.charAt(i));
        } else {
            answer[i] = -1;
        }

        map.set(s.charAt(i), i);
    }

    return answer;
}